clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -305,6 +305,13 @@ def err_drv_invalid_int_value : Error<"invalid integral value '%1' in '%0'">;
 def err_drv_invalid_value_with_suggestion : Error<
     "invalid value '%1' in '%0', expected one of: %2">;
 def err_drv_alignment_not_power_of_two : Error<"alignment is not a power of 2 in '%0'">;
+
+def err_drv_invalid_atomic_option : Error<
+  "invalid argument '%0' to -fatomic=; must be a "
```

Comment:
```suggestion
  "invalid argument '%0' to '-fatomic='; must be a "
```
Oof, this diagnostic is extremely long; perhaps this should be split into diagnostics via a `%select`?

e.g.,

`invalid argument '%0' to '-fatomic='; %select{must be a comma-separate list of key/value pairs|allowed key are <list>|allowed values are 'on' or 'off'|each key must be unique}1`

so that users don't have as much text to parse to understand what the issue is?

Reply:
will do